### PR TITLE
Apply Jetpack cloud base styles to landing page

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

Jetpack cloud has some base styles that are imported in the sidebar component. The landing page doesn't have a sidebar and therefore these styles aren't applied there. This PR fixes this by importing styles conditionally in the main layout component.

It is a follow up of #51139, which has been reverted because it inadvertently modified styles in Calypso blue.

This update makes base components have the same styles across the platform, which will make fixing contrast issues easier in a next PR.

### Implementation notes

#51139 moved the loading of the styles from the sidebar to the masterbar (which is used in the landing page), but that was just moving the problem. This PR goes a step further by conditionally loading base styles in cloud only, from the main layout component. Doing a conditional loading client side is not an issue in our case since none of cloud sections are rendered server side.

This PR touches the main layout only, which means that some pages like the login page or the 404 page won't load cloud base styles at the moment. If this implementation is accepted, we can consider porting it to those sections as well.

### Testing instructions

Download the PR and run both Calypso and cloud

In Calypso

- Browse some sections in Calypso and check that they look like production (specifically, visit `/reader` since the first PR introduced styling issues in that page)

In Jetpack cloud

- Visit the landing page
- Check that the 3 states mentioned below match production (to easily simulate these states, edit the `noSite` middleware locally at `client/jetpack-cloud/controller.ts`). Note that the input placeholder and secondary button match the rest of the site rather than the landing page in production.
- Browse some other sections in cloud and check that they look like production

### Screenshots

_No site_
111485540-131e2780-870d-11eb-8081-cbc35b75c28e

_No visible site_
111487332-a572fb00-870e-11eb-834b-b89171389a45

_1+ site_
111487302-9e4bed00-870e-11eb-805b-6e8930c662a4

### CODE REVIEW

File: client/assets/stylesheets/jetpack-cloud.scss [hunk 1]
```diff
@@ -1,32 +0,0 @@
-// Color Schemes
```
Comment (monsieur-z) [inline client/assets/stylesheets/jetpack-cloud.scss:1]: This file is not imported anywhere.

File: client/layout/index.jsx [hunk 2]
```diff
@@ -245,6 +246,9 @@ class Layout extends Component {
 				{ this.renderMasterbar() }
 				{ config.isEnabled( 'support-user' ) && <SupportUser /> }
 				<LayoutLoader />
+				{ isJetpackCloud() && (
+					<AsyncLoad require="calypso/jetpack-cloud/style" placeholder={ null } />
+				) }
```
Comment (rcanepa) [inline client/layout/index.jsx:251]: I like very much this way of loading the styles conditionally.

**Review by rcanepa — APPROVED** [review]
LGTM. I paid close attention to Calypso Blue and I didn't see any regression. Things look as expected in Calypso Green.

**Comment by monsieur-z** [comment]
I rebased with trunk and fixed styles in the pricing page. I've checked it against production, and updates are limited to the pricing page.

**Pull request merged**